python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -79,3 +93,76 @@ cdef class Scalar:
         s.c_obj.swap(libcudf_scalar)
         s._data_type = DataType.from_libcudf(s.get().type())
         return s
+
+    @classmethod
+    def from_py(cls, py_val):
+        """
+        Convert a Python standard library object to a Scalar.
+
+        Parameters
+        ----------
+        py_val: bool, int, float, str, datetime.datetime, datetime.timedelta, list, dict
+            Value to convert to a pylibcudf.Scalar
+
+        Returns
+        -------
+        Scalar
+            New pylibcudf.Scalar
+        """
+        return _from_py(py_val)
+
+cdef Scalar _new_scalar(unique_ptr[scalar] c_obj, DataType dtype):
+    cdef Scalar s = Scalar.__new__(Scalar)
+    s.c_obj.swap(c_obj)
+    s._data_type = dtype
+    return s
+
+
+@singledispatch
+def _from_py(py_val):
+    raise TypeError(f"{type(py_val).__name__} cannot be converted to pylibcudf.Scalar")
+
+
+@_from_py.register(dict)
+@_from_py.register(list)
+@_from_py.register(datetime.datetime)
+@_from_py.register(datetime.timedelta)
+def _(py_val):
+    raise NotImplementedError(
+        f"Conversion from {type(py_val).__name__} is currently not supported."
+    )
+
+
+@_from_py.register(float)
+def _(py_val):
+    cdef DataType dtype = DataType(type_id.FLOAT64)
+    cdef unique_ptr[scalar] c_obj = make_numeric_scalar(dtype.c_obj)
+    (<numeric_scalar[double]*>c_obj.get()).set_value(py_val)
+    cdef Scalar slr = _new_scalar(move(c_obj), dtype)
```

Comment:
You could probably put this call into the Scalar.from_py implementation so as to not repeat it for each overload, but then you'd also probably have to cast the inputs so it may not be any cleaner (if at all). Just a thought.

Reply:
I'll might tackle this in a future PR where I add functionality to convert numpy scalars to `pylibcudf.Scalar`s where I probably will duplicate this call even more